src/main/java/org/jboss/tools/intellij/openshift/tree/application/ApplicationTreeModel.java

```diff
@@ -120,21 +88,18 @@ public void moduleRemoved(@NotNull Project project, @NotNull Module module) {
         removeContext(getModuleRoot(module));
     }
 
-    private void addContextToSettings(String path, LocalConfig.ComponentSettings componentSettings) {
+    private void addContextToSettings(String path, ComponentDescriptor descriptor) {
         if (!components.containsKey(path)) {
-            components.put(path, new ComponentDescriptor(componentSettings.getProject(), componentSettings.getApplication(), path, componentSettings.getName()));
+            components.put(path, descriptor);
             refresh();
         }
     }
 
     private void addContext(VirtualFile modulePathFile) {
         if (modulePathFile != null && modulePathFile.isValid()) {
             try {
-                VirtualFile file = modulePathFile.findFileByRelativePath(ODO_CONFIG_YAML);
-                if (file != null && file.isValid()) {
-                    LocalConfig config = LocalConfig.load(new File(file.getPath()).toPath().toUri().toURL());
-                    addContextToSettings(modulePathFile.getPath(), config.getComponentSettings());
-                }
+                List<ComponentDescriptor> descriptorss = ROOT.getOdo().discover(modulePathFile.getPath());
```

Comment:
is this double S intended ? if no, please correct the typo.

Reply:
Good catch